Sofa/GL/CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
 cmake_minimum_required(VERSION 3.22)
 project(Sofa.GL LANGUAGES CXX)
 
+# Detect if component is compiled outside SOFA
+if ("${CMAKE_PROJECT_NAME}" STREQUAL "${PROJECT_NAME}")
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/Modules")
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../framework/Config/cmake")
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../framework/Config/cmake/Modules")
+    include(SofaMacros)
+endif()
```

Comment:
It looks quite ugly to me 😶‍🌫️
if you do `find_package(Sofa.Config REQUIRED)` it will automatically include these cmake mdules into the  CMAKE_MODULE_PATH and include SofaMacros.
That is how it is generally done in the external plugins